lib/glimesh_web/controllers/user_settings_controller.ex

```diff
@@ -77,6 +78,49 @@ defmodule GlimeshWeb.UserSettingsController do
     end
   end
 
+  def update_tfa(conn, %{
+        "current_password" => password,
+        "user" => %{"tfa" => pin},
+        "tfa_tmp" => secret
+      }) do
+    user = conn.assigns.current_user
+
+    if user.tfa_token do
+      case Accounts.update_tfa(user, pin, %{tfa_token: nil}) do
+        {:ok, user} ->
+          conn
+          |> put_flash(:info, "2FA updated successfully.")
+          |> put_session(:user_return_to, Routes.user_settings_path(conn, :edit))
+          |> UserAuth.log_in_user(user)
+
+        {:error, changeset} ->
+          render(conn, "edit.html", tfa_changeset: changeset)
+      end
+    else
```

Comment:
These two case statements look similar, the only difference I see is the `tfa_token` param. You can do functional pattern matching, or since `user.tfa_token` is already nil pass it in directly

Reply:
what I can do is switch how the if statement is as its only for removing the hash if the user already has a 2fa setup when they re enter the pin